File: Using_CSharp/behavioral/Ducks/MiniDuckSimulator.cs
using System;

namespace Ducks.Ducks {
  class MiniDuckSimulator {
    public static void Main(string[] args) {

      Duck mallard = new MallardDuck();
      Console.WriteLine("==================\nI'm a Mallard:");
      mallard.PerformQuack();
      mallard.PerformFly();

      // Setting behavior dynamically:
      Duck model = new ModelDuck();
      Console.WriteLine("==================\nI'm a Model Duck:");
      // Use Fly behavior currently set in the code:
      model.PerformFly();

      // Now set a new Fly behavior dynamically:
      model.SetFlyBehavior(new FlyRocketPowered());
      model.PerformFly();

    }
  }
}

File: using_python/behavioral/ducks/mini_duck_simulator.py
#!/usr/bin/env python3

from ducks import MallardDuck, ModelDuck
from fly_behaviors import FlyRocketPowered


def mini_duck_simulator():

    mallard_duck = MallardDuck()
    mallard_duck.display()
    mallard_duck.perform_fly()
    mallard_duck.perform_quack()

    print("\n")

    model_duck = ModelDuck()
    model_duck.display()
    model_duck.perform_fly()
    model_duck.set_fly_behavior(FlyRocketPowered())
    model_duck.perform_fly()
    model_duck.perform_quack()


if __name__ == '__main__':
    mini_duck_simulator()

